presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -114,10 +114,17 @@ public TopNNode map(TopNNode node, PlanNode source, PlanNodeId newNodeId)
     {
         ImmutableList.Builder<Symbol> symbols = ImmutableList.builder();
         ImmutableMap.Builder<Symbol, SortOrder> orderings = ImmutableMap.builder();
+        Set<Symbol> seenCanonicals = new HashSet<>(node.getOrderBy().size());
```

Comment:
Maybe just use `HashMap` instead `ImmutableMap.Builder` for `orderings` and use `Map.hasKey` to check if entry is already present.